File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.Arrays;
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int count = 0;
        Arrays.sort(citations);
        for(int i = 1; i < citations.length ; i++){
            count = 0;
            for(int j = citations.length-1; j >= 0; j--){
                if(citations[j] > i){
                    count++;
                }
            }
            if(count == i){
                answer = count;
                break;
            }else{
                answer = count;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.kt
class Solution {
    fun solution(citations: IntArray): Int {
        var answer = 0
        var count = 0
        citations.sort()
        for (i in 1 until citations.size) {
            count = 0
            for (j in citations.indices.reversed()) {
                if (citations[j] > i) {
                    count++
                }
            }
            if (count == i) {
                answer = count
                break
            } else {
                answer = count
            }
        }
        return answer
    }
}